### PR TITLE
Added TagReader class to read tags from ItemStack

### PR DESCRIPTION
**Pull request by Aleksey-Terzi**
TagReader can read NBT tags from ItemStack.
It is used in Printing Press recipe classes in FactoryMod.

### CODE REVIEW

**Comment by Maxopoly** [comment]
I think while you are on it, it would be a good idea to add setters as well. The existing stuff for ItemMaps is directly tied to config parsing, having an API to set/get all that nbt stuff in one place would be nice.

File: src/main/java/vg/civcraft/mc/civmodcore/itemHandling/TagReader.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import net.minecraft.server.v1_10_R1.NBTTagCompound;
+import net.minecraft.server.v1_10_R1.NBTTagList;
+
+import org.bukkit.craftbukkit.v1_10_R1.inventory.CraftItemStack;
+import org.bukkit.inventory.ItemStack;
+
+public class TagReader {
+	private NBTTagCompound tag;
+	
+	public TagReader(ItemStack is) {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/itemHandling/TagReader.java]: I'd do a null check here and throw an IllegalArgumentException if the given stack is null

**Comment by ProgrammerDan** [comment]
I assume that the intent is to expand this as needs for more NBT features arise?

**Comment by Aleksey-Terzi** [comment]
Yes, I supposed this is to be simple class just to satisfy needs of my classes for FactoryMod and could be extended when someone will need this.
But since "someone" already need this - I will do it more universal and with "write" support.
Class will be renamed to TagManager.

**Comment by Aleksey-Terzi** [comment]
> I'd do a null check here and throw an IllegalArgumentException if the given stack is null

This is done.

Also class TagReader was replaced by TagManager which allows to set values of some types.
FactoryMod now is using TagManager to write values to nbt tag.

**Comment by Maxopoly** [comment]
Nice work, should be merged imo.

**Comment by ProgrammerDan** [comment]
Looks great

**Pull request merged**